runtime/script_loader.cpp

```diff
@@ -137,6 +139,99 @@ static JSObject* get_module(JSContext* cx, const char* specifier, const char* re
   return module;
 }
 
+static JSObject* get_builtin_module(JSContext* cx, HandleValue id, HandleObject builtin) {
```

Comment:
This does a synthetic module generation via source text. For example, for a JS builtin object like `{ env }`, the generated code looks like:

```js
const { e0: 'env' } = import.meta.builtin;
export { e0 as 'env' }
```

By always quoting the strings both in the destructuring and the reexport, we ensure all export names are supported.

Strictly speaking, this code doesn't yet support `'` in the export key currently, I can add that if we want to ensure 100% correctness.